### ISSUE TITLE
operator olm gui  error Invariant Violation while accessing csi scale operator panel

### ISSUE THREAD

**Issue by gandhisanjayv** [IBM/ibm-spectrum-scale-csi#70]
**Describe the bug**
Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=object%20with%20keys%20%7Bk8sNode%2C%20spectrumscaleNode%7D&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings. 

```
in span
    in Unknown
    in dd
    in dl
    in N
    in div
    in div
    in div
    in div
    in div
    in Unknown
    in C
    in Unknown
    in t
    in t
    in div
    in Data
    in StatusBox
    in div
    in div
    in Unknown
    in Unknown
    in Unknown
    in Unknown
    in C
    in t
    in C
    in Unknown
    in t
    in DetailsPage
    in OperandDetailsPage
    in t
    in Unknown
    in C
    in t
    in t
    in div
    in div
    in section
    in d
    in Unknown
    in withRouter(undefined)
    in C
    in main
    in div
    in y
    in t
    in t
    in t
    in t
a/<@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:425
a@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:528
po@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:47246
p@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:48881
m@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:49983
ho/<@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:51959
wi@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:58902
Mi@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:67287
Va@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:90685
qa@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:91069
ku@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:97956
Au@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:97336
Pu@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:104:98430
notify@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:22:26329
a</t.notifyNestedSubs@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:22:25899
f</n.notifySubscribers@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:44:176151
a</t.handleChangeWrapper@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:22:25967
v@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:22:29187
d/</</<@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/main-chunk-a4083cec3da7373f9e27.min.js:1:200253
dispatch@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:22:32686
l/<@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/main-chunk-a4083cec3da7373f9e27.min.js:1:5837
a@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:88:32986
j/<@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:88:33108
l@https://console-openshift-console.apps.viserion.scale-ocp.tuc.stglabs.ibm.com/static/vendors~main-chunk-ae0a1c71d377795aeb13.min.js:88:35744
```

**To Reproduce**
edit operator from command line and add node mapping -> try to access from GUI

```
 oc edit CSIScaleOperator ibm-spectrum-scale-csi  -n  ibm-spectrum-scale-csi-driver
```

**Expected behavior**
should able to access from GUI

**Environment**
openshift 4.2
# Developement. 
GA version v1.0.0

# Deployment
OpenShift version
    4.2.0
**Screenshots**

**Additional context**
i was not able to add node mapping from gui, It was failing, I did not save that error.

**Comment by mew2057** [comment]
Can I see the output of `oc get CSIScaleOperator ibm-spectrum-scale-csi  -n  ibm-spectrum-scale-csi-driver -o yaml`? 

This is likely something that we  have zero  control over, this is likely an OLM issue that we can't solve from the operator. My guess is an bad character somehow got mixed in. My OLM environment is currently down, but I'll prioritize this as a point of investigation.

**Comment by gandhisanjayv** [comment]
oc get CSIScaleOperator ibm-spectrum-scale-csi -n ibm-spectrum-scale-csi-driver -o yaml
apiVersion: csi.ibm.com/v1
kind: CSIScaleOperator
metadata:
  creationTimestamp: "2020-02-11T18:21:16Z"
  finalizers:
  - finalizer.csiscaleoperators.csi.ibm.com
  generation: 16
  labels:
    app.kubernetes.io/instance: ibm-spectrum-scale-csi-operator
    app.kubernetes.io/managed-by: ibm-spectrum-scale-csi-operator
    app.kubernetes.io/name: ibm-spectrum-scale-csi-operator
  name: ibm-spectrum-scale-csi
  namespace: ibm-spectrum-scale-csi-driver
  resourceVersion: "36104821"
  selfLink: /apis/csi.ibm.com/v1/namespaces/ibm-spectrum-scale-csi-driver/csiscaleoperators/ibm-spectrum-scale-csi
  uid: 4aae5220-4cfb-11ea-b844-00505695e242
spec:
  clusters:
  - id: "14225222090139407765"
    primary:
      primaryFs: fs1
      primaryFset: csifset-olm-feb11
    restApi:
    - guiHost: viserion-compute6.viserion.scale-ocp.tuc.stglabs.ibm.com
    secrets: guisecret
    secureSslMode: false
  nodeMapping:
  - k8sNode: viserion-compute1
    spectrumscaleNode: viserion-compute1.viserion.scale-ocp.tuc.stglabs.ibm.com
  - k8sNode: viserion-compute2
    spectrumscaleNode: viserion-compute2.viserion.scale-ocp.tuc.stglabs.ibm.com
  - k8sNode: viserion-compute3
    spectrumscaleNode: viserion-compute3.viserion.scale-ocp.tuc.stglabs.ibm.com
  - k8sNode: viserion-compute4
    spectrumscaleNode: viserion-compute4.viserion.scale-ocp.tuc.stglabs.ibm.com
  scaleHostpath: /ibm/fs1
  secretCounter: 12
status:
  conditions:
  - ansibleResult:
      changed: 0
      completion: 2020-02-12T16:34:27.132634
      failures: 0
      ok: 14
      skipped: 2
    lastTransitionTime: "2020-02-11T18:21:16Z"
    message: Awaiting next reconciliation
    reason: Successful
    status: "True"
    type: Running

**Comment by mew2057** [comment]
Okay I think it's in the CSV now  that I look at  it:  

`- urn:alm:descriptor:com.tectonic.ui:label`

we assign the Spectrum Scale Node a label, and  I think that  react is not happy about that.  The workaround for this is to click  the three dots on the CR in  the GUI then click `Edit CSIScale Operator`

![Screen Shot 2020-02-12 at 12 01 17 PM](https://user-images.githubusercontent.com/1195452/74358516-8da0a900-4d8f-11ea-9263-ed9ed8ad5d46.png)

**Comment by gandhisanjayv** [comment]
above procedure worked.

**Comment by mew2057** [comment]
Resolved in #72.
